src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocLeadingAsteriskAlignCheck.java

```diff
@@ -0,0 +1,240 @@
+        // this method checks the alignment of leading asterisks.
+        final boolean isJavadocStartingLine = ast.getLineNumber() == javadocStartLineNumber;
+        if (!isJavadocStartingLine) {
+            final OptionalInt asteriskColumnNumber =
+                                        getAsteriskColumnNumberWithTabsExpanded(ast.getText());
+
+            asteriskColumnNumber.ifPresent(integer -> {
+                checkAsteriskAlignment(ast.getLineNumber(),
+                        expectedColumnNumber,
+                        integer);
+            });
+        }
+    }
+
+    @Override
+    public void finishJavadocTree(DetailNode rootAst) {
+        // this method checks the alignment of closing javadoc tag.
+        final DetailAST javadocEndToken = getBlockCommentAst().getLastChild();
+        final String lastLine = fileLines[javadocEndToken.getLineNo() - 1];
+        final OptionalInt asteriskColumnNumber =
+                    getAsteriskColumnNumberWithTabsExpanded(lastLine);
+
+        asteriskColumnNumber.ifPresent(integer -> {
+            checkAsteriskAlignment(javadocEndToken.getLineNo(),
+                    expectedColumnNumber,
+                    integer);
+        });
+    }
+
+    /**
+     * Processes and returns an Optional containing
+     * the column number of leading asterisk with tabs expanded.
+     * Also sets 'singleSpaceAsteriskColNumber' if the leading asterisk is present.
+     *
+     * @param line single line from block comment text
+     * @return column number of leading asterisk with tabs expanded
+     */
+    private OptionalInt getAsteriskColumnNumberWithTabsExpanded(String line) {
+        OptionalInt columnNumber = getAsteriskColumnNumber(line);
+
+        if (columnNumber.isPresent()) {
+            singleSpaceAsteriskColNumber = columnNumber.orElseThrow();
+            columnNumber = OptionalInt.of(CommonUtil.lengthExpandedTabs(
+                        line, columnNumber.orElseThrow(), getTabWidth()));
+        }
+
+        return columnNumber;
+    }
+
+    /**
+     * Processes and returns an Optional containing
+     * the column number of leading asterisk without tabs expanded.
+     *
+     * @param line javadoc comment line
+     * @return asterisk's column number
+     */
+    private static OptionalInt getAsteriskColumnNumber(String line) {
+        final Pattern pattern = Pattern.compile("^(\\s*)\\*");
+        final Matcher matcher = pattern.matcher(line);
+        final OptionalInt columnNumber;
+
+        // We may not always have a leading asterisk because a javadoc line can start with
+        // a non-whitespace character or the javadoc line can be empty.
+        // In such cases, there is no leading asterisk and Optional will be empty.
+        if (matcher.find()) {
+            columnNumber = OptionalInt.of(matcher.group(1).length() + 1);
+        }
+        else {
+            columnNumber = OptionalInt.empty();
+        }
+
+        return columnNumber;
```

Comment:
It is likely that we can make checker happier and avoid thinking of a better name for `columnNumber` by directly returning a value here:
```suggestion
        return Optional.of(matcher.group(1))
                .map(l -> l.length() + 1)
                .map(OptionalInt::of)
                .orElse(OptionalInt.empty());
```

Reply:
Done. here is the new changes:

https://github.com/checkstyle/checkstyle/blob/77dd1d104385ad79133495efee9f504f23f06730/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocLeadingAsteriskAlignCheck.java#L208-L213

needed to add `.filter(Matcher::find) ` because test cases were not running without it.

And `.orElse(OptionalInt.empty());` is causing build failure, here is the error ( spotbugs error ):

```
[ERROR] Medium: Method com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocLeadingAsteriskAlignCheck.getAsteriskColumnNumber(String) uses immediate execution of a block of cod
e that is often not used [com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocLeadingAsteriskAlignCheck] At JavadocLeadingAsteriskAlignCheck.java:[line 213] OI_OPTIONAL_ISSUES_USES_IMMEDIATE_EXECUTION
```

Reply:
You need to use orElseGet

Reply:
@mahfouz72 do you remember which PR I shared a good explanation of this concept in? I can't find it.

Reply:
@nrmancuso https://github.com/checkstyle/checkstyle/pull/14623#issuecomment-2004294036

Reply:
had to add this suppression because of orElseGet:

https://github.com/checkstyle/checkstyle/blob/4c812934107632c64377b91dc685c1a082c02fce/config/checker-framework-suppressions/checker-nullness-optional-interning-suppressions.xml#L3941-L3946